packages/php-wasm/web/src/lib/web-php.ts

```diff
@@ -44,6 +48,560 @@ const fakeWebsocket = () => {
 	};
 };
 
+async function generateKeys() {
+	const keyPair = await crypto.subtle.generateKey(
+		{
+			name: 'ECDH',
+			namedCurve: 'P-256',
+		},
+		true,
+		['deriveKey']
+	);
+	return keyPair;
+}
+
+function createCertificate({ commonName }) {
+	// create certificate
+	var pki = forge.pki;
+	var keys = pki.rsa.generateKeyPair(2048);
+	var cert = pki.createCertificate();
+	cert.publicKey = keys.publicKey;
+	// alternatively set public key from a csr
+	//cert.publicKey = csr.publicKey;
+	// NOTE: serialNumber is the hex encoded value of an ASN.1 INTEGER.
+	// Conforming CAs should ensure serialNumber is:
+	// - no more than 20 octets
+	// - non-negative (prefix a '00' if your value starts with a '1' bit)
+	cert.serialNumber = '01';
+	cert.validity.notBefore = new Date();
+	cert.validity.notAfter = new Date();
+	cert.validity.notAfter.setFullYear(
+		cert.validity.notBefore.getFullYear() + 1
+	);
+	var attrs = [
+		{
+			name: 'commonName',
+			value: commonName,
+		},
+		{
+			name: 'countryName',
+			value: 'US',
+		},
+		{
+			shortName: 'ST',
+			value: 'Virginia',
+		},
+		{
+			name: 'localityName',
+			value: 'Blacksburg',
```

Comment:
we probably don't need all of these attributes do we? if not, maybe we should have a good explanation for why we're using the very specific values we are. we could probably make up a virtual/imaginary Playground land in WordPressWorld so that if someone _were_ to examine the certificates it would be clearer that these are for a virtual system and not supposed to be actual legitimate signed certificates